api/base/parsers.py

```diff
@@ -59,7 +59,7 @@ class JSONAPIParser(JSONParser):
     @staticmethod
     def get_relationship(data, related_resource):
         target_type = data.get('type')
-        if not target_type:
+        if not target_type and data:
```

Comment:
If you update to an empty list to remove all affilations, don't need to include `type`